hs_app_netCDF/tests/test_netcdf_metadata.py

```diff
@@ -1,179 +1,534 @@
+        self.assertEquals(self.resNetcdf.metadata.variables.all().count(), 2)
+
+        # delete
+        # variable deletion is not allowed
+        with self.assertRaises(ValidationError):
+            self.resNetcdf.metadata.delete_element('variable', self.resNetcdf.metadata.variables.all().filter(name='SWE').first().id)
+
+        # update
+        # update original coverage element
+        value_2 = {"northlimit": '12.5', "projection": "transverse_mercator", "units": "meter", "southlimit": '10.5',
+            "eastlimit": '23.5', "westlimit": '2.5'}
+        self.resNetcdf.metadata.update_element('originalcoverage',
+                                               ori_coverage.id,
+                                               value=value_2,
+                                               projection_string_text='2009',
+                                               projection_string_type='EPSG Code'
+                                               )
+        ori_coverage = self.resNetcdf.metadata.ori_coverage.all().first()
+        self.assertEquals(ori_coverage.value, value_2)
+        self.assertEquals(ori_coverage.projection_string_text, '2009')
+        self.assertEquals(ori_coverage.projection_string_type, 'EPSG Code')
+
+        # update variable element
+        variable = self.resNetcdf.metadata.variables.all().filter(name='SWE').first()
+        self.resNetcdf.metadata.update_element('variable', variable.id,
+                                                   name='SWE2', type='Double',
+                                                   shape='y,x,time', unit='m',
+                                                   missing_value='-999', descriptive_name='snow water equivalent',
+                                                   method='model result of UEB')
+        variable = self.resNetcdf.metadata.variables.all().filter(name='SWE2').first()  # need to refer to the variable again!
+        self.assertEquals(variable.name, 'SWE2')
+        self.assertEquals(variable.type, 'Double')
+        self.assertEquals(variable.shape, 'y,x,time')
+        self.assertEquals(variable.unit, 'm')
+        self.assertEquals(variable.missing_value, '-999')
+        self.assertEquals(variable.descriptive_name, 'snow water equivalent')
+        self.assertEquals(variable.method, 'model result of UEB')
+
+    def test_get_xml(self):
+        # add a valid netcdf file to generate metadata
+        files = [UploadedFile(file=self.netcdf_file_obj, name=self.netcdf_file_name)]
+        utils.resource_file_add_pre_process(resource=self.resNetcdf, files=files, user=self.user,
+                                            extract_metadata=False)
+
+        utils.resource_file_add_process(resource=self.resNetcdf, files=files, user=self.user,
+                                        extract_metadata=True)
+
+        # test if xml from get_xml() is well formed
+        ET.fromstring(self.resNetcdf.metadata.get_xml())
```

Comment:
This function currently just tests whether the xml is well formed. We plan to add more code in this function to compare the value from the xml and from the resource metadata element object.  

Is it possible to keep this function as it is and add more code for this function in the future? Otherwise, I will remove it for current pull request. Thanks